### PR TITLE
Put items from the back-end in the front-end

### PR DESCRIPTION
**Pull request by santiago-a-serrano**

### CODE REVIEW

**Review by thienantran010 — APPROVED** [review]
good work with item template

File: project/src/main/webapp/results.js [hunk 1]
```diff
@@ -48,21 +48,46 @@ function addCollapsibleEffect() {
...
+    const itemTemplate = await fetch('/item_template.html')
+    const templateText = await itemTemplate.text()
+    //General items:
+    const generalItemsHere = document.getElementById("general-item-list")
+    var generalItemsHTML = generalItemsHere.innerHTML
+    for(let i = 0; i < jsonResponse.generalItems.length; i++){
+        const item = jsonResponse.generalItems[i]
+        generalItemsHTML += replaceItemHTML(templateText, "general", item.name, item.explanation);
+    }
+    generalItemsHere.innerHTML = generalItemsHTML
```
Comment (thienantran010) [inline project/src/main/webapp/results.js:69]: line 64 says generalItemsHTML = generalItemsHere.innerHTML so isn't line 69 unnecessary

File: project/src/main/webapp/results.js [hunk 2]
```diff
@@ -48,21 +48,46 @@ function addCollapsibleEffect() {
...
+    //Course-specific items:
+    const courseItemsHere = document.getElementById("course-item-list")
+    var courseItemsHTML = courseItemsHere.innerHTML
+    for(let i = 0; i < jsonResponse.majorItems.length; i++){
+        const item = jsonResponse.majorItems[i]
+        courseItemsHTML += replaceItemHTML(templateText, "course", item.name, item.explanation);
+    }
+    courseItemsHere.innerHTML = courseItemsHTML
```
Comment (thienantran010) [inline project/src/main/webapp/results.js:77]: same issue (from line 64 and 69) here but with courseItemsHere and courseItemsHTML

File: project/src/main/webapp/results.js [hunk 3]
```diff
@@ -48,21 +48,46 @@ function addCollapsibleEffect() {
...
+    document.getElement
```
Comment (thienantran010) [inline project/src/main/webapp/results.js:78]: i think this is a typo

File: project/src/main/webapp/results.js [hunk 4]
```diff
@@ -48,21 +48,46 @@ function addCollapsibleEffect() {
...
+
+    /*Add collapsing effect event to every button as soon as it loads*/
+    addCollapsibleEffect()
 }
 
-/*Add collapsing effect event to every button as soon as it loads*/
-window.onload = addCollapsibleEffect;
+//Gets the HTML that must be put for an item to be displayed
+function replaceItemHTML(templateText, itemType, itemName, itemExplanation){
```
Comment (thienantran010) [inline project/src/main/webapp/results.js:85]: very cool

**Comment by thienantran010** [comment]
ok i approved it prematurely lol but yeah if we can resolve some stuff we can get it movin

File: project/src/main/webapp/results.js (hunk 3 above)
Comment (thienantran010) [reply project/src/main/webapp/results.js:78]: line 78

**Comment by santiago-a-serrano** [comment]
@thienantran010 solved the issues you commented and amended the commit. Lmk if there's anything else or if we can merge now

**Pull request merged**